### PR TITLE
feat(langgraph): Allow option to prevent subgraphs inheriting checkpointer

### PR DESCRIPTION
**Pull request by jacoblee93**
@hinthornw

### CODE REVIEW

File: libs/langgraph/src/pregel/loop.ts [hunk 1]
```diff
@@ -311,7 +315,13 @@ export class PregelLoop {
       // Start the store. This is a batch store, so it will run continuously
       store.start();
     }
-
+    if (checkSubgraphs && isNested && params.checkpointer !== undefined) {
+      if (_SEEN_CHECKPOINT_NS.has(config.configurable?.checkpoint_ns)) {
+        throw new MultipleSubgraphsError();
```
Comment (hinthornw) [inline libs/langgraph/src/pregel/loop.ts:320]: Would be nice to have a better error message here (if we know the name where this is occurring, recommendation on what to do)

File: libs/langgraph/src/errors.ts [hunk 2]
```diff
@@ -97,3 +97,20 @@ export class InvalidUpdateError extends Error {
...
+  }
+
+  static get unminifiable_name() {
+    return "MultipleSubgraphError";
+  }
+}
+
+/**
+ * Used for subgraph detection.
+ * TODO: Move to globalThis for bundlers
```
Comment (hinthornw) [inline libs/langgraph/src/errors.ts:114]: ooc: what's blocking us from doing now?

File: libs/langgraph/src/pregel/retry.ts [hunk 3]
```diff
@@ -167,6 +167,16 @@ async function _runWithRetry(
           2
         )} seconds (attempt ${attempts}) after ${errorName}: ${error}`
       );
+      // Clear checkpoint_ns seen (for subgraph detection)
+      const checkpointNs = pregelTask.config?.configurable?.checkpoint_ns;
+      if (checkpointNs) {
+        _SEEN_CHECKPOINT_NS.delete(checkpointNs);
```
Comment (hinthornw) [inline libs/langgraph/src/pregel/retry.ts:173]: does this raise an error if not found?

**Review by hinthornw — COMMENTED** [review]
thank you! I think it makes sense? Any tests we're missing?

File: libs/langgraph/src/errors.ts (hunk 2 above)
Comment (jacoblee93) [reply libs/langgraph/src/errors.ts:114]: Feels premature and don't want to rely on globals if possible, maybe would be a good idea.

File: libs/langgraph/src/pregel/retry.ts (hunk 3 above)
Comment (jacoblee93) [reply libs/langgraph/src/pregel/retry.ts]: No, should be idempotent:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/delete

**Pull request merged**